### ISSUE TITLE
Optimize block processing

### ISSUE THREAD

**Issue by roman-khimov** [nspcc-dev/neo-go#2031]
We're very slow at (mostly empty) block processing on N3:
```
2021-06-30T11:57:59.210+0300    INFO    persisted to disk       {"blocks": 42, "keys": 2447, "headerHeight": 227962, "blockHeight": 227961, "took": "18.095386ms"}
2021-06-30T11:58:00.215+0300    INFO    persisted to disk       {"blocks": 44, "keys": 2663, "headerHeight": 228006, "blockHeight": 228005, "took": "5.827038ms"}
2021-06-30T11:58:01.237+0300    INFO    persisted to disk       {"blocks": 40, "keys": 2381, "headerHeight": 228046, "blockHeight": 228045, "took": "21.396159ms"}
2021-06-30T11:58:02.252+0300    INFO    persisted to disk       {"blocks": 56, "keys": 1678, "headerHeight": 228102, "blockHeight": 228101, "took": "14.162552ms"}
2021-06-30T11:58:03.260+0300    INFO    persisted to disk       {"blocks": 56, "keys": 1326, "headerHeight": 228158, "blockHeight": 228157, "took": "7.777113ms"}
2021-06-30T11:58:04.269+0300    INFO    persisted to disk       {"blocks": 56, "keys": 1323, "headerHeight": 228214, "blockHeight": 228213, "took": "7.916556ms"}
```
It starts at about 500-600 blocks per second and then drops to these numbers. Of course N3 does much more with each block, but we need to see if anything can be optimized there.

**Comment by fyrchik** [comment]
I have already encountered this today, caching value for `gasPerBlock` improves throughput a lot. Will make a PR later.

**Comment by fyrchik** [comment]
Sorry, `gasPerVote`, of course, which is updated for multiple keys in every `PostPersist`.

**Comment by fyrchik** [comment]
I tested restore after #2032 with current testnet dump:
On the start speed is 1k-2k blocks per second, with half of the time spent in block verification.
At around 238k block it drops to ~150 BPS for some time but then increases back to 1.5k.

Pprof tells that at that time there are some `storage.Seek` invocations for `GetStorageItemsWithPrefix`. I have created a separate task for that.
